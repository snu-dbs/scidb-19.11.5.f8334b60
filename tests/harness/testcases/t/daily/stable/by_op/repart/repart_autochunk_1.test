# Test repart() autochunking with a target physical chunk size.

--setup
# Store half a million strings of size 201 in a 2-D array.
create temp array A0 <s:string>[i=0:400:0:1000; j=0:400:0:1000]
--igdata "store(redimension(apply(build(<s:string>[row=0:80399], '/.' * 100), i, row / 400, j, row % 400), A0), A0)"
_show_nodist(A0)

--test

--echo Repartition to 3 MiB physical chunk sizes.  89 * 178 * 201 == 3,184,242
--igdata "store(repart(A0, <s:string>[i=0:400; j=0:400], phys_chunk_size: 3), A1)"
_show_nodist(A1)

--echo Repartition to 16 MiB physical chunk sizes.  201 * 400 * 201 == 16,160,400
--igdata "store(repart(A0, <s:string>[i=0:400; j=0:400], phys_chunk_size: 16), A2)"
_show_nodist(A2)

# Check expected error conditions.
--error --code=scidb::SCIDB_SE_INFER_SCHEMA::SCIDB_LE_PARAMETER_NOT_POSITIVE_INTEGER "repart(A0, <s:string>[i=0:400; j=0:400], phys_chunk_size: 0 - 16)"
--error --code=scidb::SCIDB_SE_INFER_SCHEMA::SCIDB_LE_PARAMETER_NOT_POSITIVE_INTEGER "repart(A0, <s:string>[i=0:400; j=0:400], phys_chunk_size: 0)"
--error --code=scidb::SCIDB_SE_INFER_SCHEMA::SCIDB_LE_MUTUALLY_EXCLUSIVE_PARAMS "repart(A0, <s:string>[i=0:400; j=0:400], phys_chunk_size: 4, cells_per_chunk: 72)"

--cleanup
remove(A0)
remove(A1)
remove(A2)


