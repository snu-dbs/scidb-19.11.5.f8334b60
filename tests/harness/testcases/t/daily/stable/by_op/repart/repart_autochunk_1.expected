Query was executed successfully

[Query was executed successfully, ignoring data output by this query.]

{i} schema
{0} 'A0<s:string> [i=0:400:0:1000; j=0:400:0:1000]'

Repartition to 3 MiB physical chunk sizes.  89 * 178 * 201 == 3,184,242
[Query was executed successfully, ignoring data output by this query.]

{i} schema
{0} 'A1<s:string> [i=0:400:0:89; j=0:400:0:178]'

Repartition to 16 MiB physical chunk sizes.  201 * 400 * 201 == 16,160,400
[Query was executed successfully, ignoring data output by this query.]

{i} schema
{0} 'A2<s:string> [i=0:400:0:201; j=0:400:0:400]'

[An error expected at this place for the query "repart(A0, <s:string>[i=0:400; j=0:400], phys_chunk_size: 0 - 16)". And it failed with error code = scidb::SCIDB_SE_INFER_SCHEMA::SCIDB_LE_PARAMETER_NOT_POSITIVE_INTEGER. Expected error code = scidb::SCIDB_SE_INFER_SCHEMA::SCIDB_LE_PARAMETER_NOT_POSITIVE_INTEGER.]

[An error expected at this place for the query "repart(A0, <s:string>[i=0:400; j=0:400], phys_chunk_size: 0)". And it failed with error code = scidb::SCIDB_SE_INFER_SCHEMA::SCIDB_LE_PARAMETER_NOT_POSITIVE_INTEGER. Expected error code = scidb::SCIDB_SE_INFER_SCHEMA::SCIDB_LE_PARAMETER_NOT_POSITIVE_INTEGER.]

[An error expected at this place for the query "repart(A0, <s:string>[i=0:400; j=0:400], phys_chunk_size: 4, cells_per_chunk: 72)". And it failed with error code = scidb::SCIDB_SE_INFER_SCHEMA::SCIDB_LE_MUTUALLY_EXCLUSIVE_PARAMS. Expected error code = scidb::SCIDB_SE_INFER_SCHEMA::SCIDB_LE_MUTUALLY_EXCLUSIVE_PARAMS.]

Query was executed successfully

Query was executed successfully

Query was executed successfully

